### ISSUE TITLE
Detect if existing .pot file needs to be updated

### ISSUE THREAD

**Issue by techtonik** [python-babel/babel#273]
Sphinx uses Babel. And while generating `.pot` file it adds timestamps. Many timestamps. Even if `.pot` file content stays the say. Is it possible to use Babel to detect if existing `.pot` file needs to be regenerated?

Started in https://github.com/sphinx-doc/sphinx/issues/2073#issuecomment-146950096

**Comment by sils1297** [comment]
Hi @techtonik , realistically nobody will work on this feature request soon because we don't have manpower to address feature requests. I'm happy to assist you if you want to implement it. At any time you can ping me with my github username on https://gitter.im/python-babel/babel , others may help you too there :).

**Comment by techtonik** [comment]
@sils1297 when Babel reads `.pot` file, is it a structure that can be easily compared? The same stuff is for building - is the structure that Babel is about to write into `.pot` is the same as the one produces when reading `.pot`s?

**Comment by sils1297** [comment]
> when Babel reads .pot file, is it a structure that can be easily compared?

Nope. See https://github.com/python-babel/babel/blob/master/babel/messages/frontend.py#L311 , you may want to implement a content equality comparison operators for the Catalog class and add an option for what you desire. I'm not sure if we want it to be active by default because it'll have a performance impact.

**Comment by techtonik** [comment]
@sils1297 looks like `write_po` should be a method in a POFile class that encompasses state and options passed as function parameters. https://github.com/python-babel/babel/blob/master/babel/messages/pofile.py#L337

**Comment by sils1297** [comment]
@techtonik well this is historically grown and nobody around has time to do a proper refactoring here.

**Comment by techtonik** [comment]
Ok. I may look into it some day.
